src/lib/snyk-to-html.ts

```diff
@@ -347,6 +351,20 @@ async function processCodeData(
 
   const OrderedIssuesArray = await processSourceCode(dataArray);
 
+  // Process suppressions
+  OrderedIssuesArray.forEach(project => {
+    project.vulnerabilities = project.vulnerabilities.map(vuln => {
+      if (vuln.suppressions && vuln.suppressions.length > 0) {
+        vuln.suppression = processSuppression(vuln.suppressions[0]);
+      }
+      return vuln;
+    }).sort((a, b) => {
+      if (a.suppression && !b.suppression) return 1;
+      if (!a.suppression && b.suppression) return -1;
+      return 0;
+    });
+  });
```

Comment:
**suggestion:** We only need to perform this sort option if one or more of the project vulnerabilities had suppression data. So we could skip this step if `hasSuppressedVulns = false` within the forEach loop. 

```suggestion
  OrderedIssuesArray.forEach(project => {
    let hasSuppressedVulns = false;
    const projectVulns = project.vulnerabilities.map(vuln => {
      if (vuln.suppressions && vuln.suppressions.length > 0) {
        hasSuppressedVulns = true;
        vuln.suppression = processSuppression(vuln.suppressions[0]);
      }
      return vuln;
    });
  
    if (!hasSuppressedVulns) {
      project.vulnerabilities = projectVulns;
      return; // Early return if no suppressions
    }
  
    // Sort only if necessary
    projectVulns.sort((a, b) => {
      if (a.suppression && !b.suppression) return 1;
      if (!a.suppression && b.suppression) return -1;
      return 0;
    });
  
    project.vulnerabilities = projectVulns;
  });
```